src/settingsScreen.js

```diff
@@ -38,53 +48,336 @@ class SettingsScreen {
+        }
+        await match.accessible.SetPickerValue(1, `${sensitivity.bgValuePerInsulinUnit} ${config.insulinSensitivitiesUnits}`);
+    }
+    /**
+     * @param {Object} sensitivity
+     * @param {String} sensitivity.time
+     * @param {String} sensitivity.bgValuePerInsulinUnit
+     */
+    async Edit(sensitivity) {
+        await match.accessible.Label(`${sensitivity.time}`).atIndex(0).tap();
+        await match.accessible.SetPickerValue(1, `${sensitivity.bgValuePerInsulinUnit} ${config.insulinSensitivitiesUnits}`);
+    }
+}
+
+class CorrectionRangeScreen {
+    constructor(language) {
+        this.language = language;
+    }
+    _pickerIndexForTime(time) {
+        switch (time) {
+            case '12:00 AM':
+                return 0;
+            case '12:30 AM':
+                return 1;
+            case '1:00 AM':
+                return 2;
+            case '1:30 AM':
+                return 3;
+            case '2:00 AM':
+                return 4;
+            case '2:30 AM':
+                return 6;
+            default:
+                return 0;
+        }
+    }
+    CorrectionRangeHeader() {
+        return match.accessible.Header(this.language.settingsScreen.CorrectionRange);
+    }
+    async Close() {
+        await match.accessible.BackButton(this.language.settingsScreen.Settings).tap();
+    }
+    async Save() {
+        await match.accessible.Label(this.language.general.Save).tap();
+    }
+    async Cancel() {
+        await match.accessible.ButtonBarButton(this.language.general.Cancel).tap();
+    }
+    async ApplyAll(ranges) {
+        for (let index = 0; index < ranges.length; index++) {
+            await this.Apply(ranges[index]);
+        }
+    }
+    /**
+     * @param {Object} range
+     * @param {String} range.time
+     * @param {String} range.max
+     * @param {String} range.min
+     */
+    async Apply(range) {
+        let pickerIndex = this._pickerIndexForTime(range.time);
+        await match.accessible.ButtonBarButton(this.language.general.Add).tap();
+        await match.accessible.Label(`${range.time}`).atIndex(pickerIndex).tap();
+        let currentMax = config.correctionRangesMaximum;
+        do {
+            await match.accessible.PickerItem(1, `${currentMax}`).tap();
+            currentMax--;
+        } while (currentMax >= range.max);
+
+        let currentMin = range.max;
+        do {
+            if (currentMin == range.max) {
+                await match.accessible.PickerItem(4, `${currentMin}`).tap();
+            } else if (currentMin == (range.max - 1)) {
+                await match.accessible.PickerItem(2, `${currentMin}`).tap();
+            } else {
+                await match.accessible.PickerItem(1, `${currentMin}`).tap();
+            }
+            currentMin--;
+        } while (currentMin >= range.min);
+    }
+}
+
+class CarbRatiosScreen {
+    constructor(language) {
+        this.language = language;
+    }
+    CarbRatiosHeader() {
```

Comment:
[nit] Isn't this redundant `CarbRatiosScreen.CarbRatiosHeader`?

Reply:
yeah - I will update those